addons/mrp/migrations/10.0.2.0/openupgrade_analysis_work.txt

```diff
@@ -0,0 +1,424 @@
+mrp          / mrp.workorder            / active_move_lot_ids (one2many): NEW relation: stock.move.lots
+mrp          / mrp.workorder            / capacity (float)              : NEW
+mrp          / mrp.workorder            / date_finished (datetime)      : NEW
+mrp          / mrp.workorder            / date_planned_finished (datetime): NEW
+mrp          / mrp.workorder            / date_planned_start (datetime) : NEW
+mrp          / mrp.workorder            / date_start (datetime)         : NEW
+mrp          / mrp.workorder            / duration_expected (float)     : NEW
+mrp          / mrp.workorder            / final_lot_id (many2one)       : NEW relation: stock.production.lot
+mrp          / mrp.workorder            / message_follower_ids (one2many): NEW relation: mail.followers
+mrp          / mrp.workorder            / message_ids (one2many)        : NEW relation: mail.message
+mrp          / mrp.workorder            / message_last_post (datetime)  : NEW
+mrp          / mrp.workorder            / move_lot_ids (one2many)       : NEW relation: stock.move.lots
+mrp          / mrp.workorder            / move_raw_ids (one2many)       : NEW relation: stock.move
+mrp          / mrp.workorder            / name (char)                   : NEW required: required
+# NOTHING TO DO
+
+mrp          / mrp.workorder            / next_work_order_id (many2one) : NEW relation: mrp.workorder
+# DONE: Ordered the workorders of a same mrp_production based on the v9 'sequence' field and the ids.
+
+mrp          / mrp.workorder            / operation_id (many2one)       : NEW relation: mrp.routing.workcenter
+mrp          / mrp.workorder            / production_id (many2one)      : NEW relation: mrp.production, required: required
+mrp          / mrp.workorder            / qty_produced (float)          : NEW
+mrp          / mrp.workorder            / qty_producing (float)         : NEW
+mrp          / mrp.workorder            / scrap_ids (one2many)          : NEW relation: stock.scrap
+mrp          / mrp.workorder            / state (selection)             : NEW selection_keys: ['cancel', 'done', 'pending', 'progress', 'ready']
+mrp          / mrp.workorder            / time_ids (one2many)           : NEW relation: mrp.workcenter.productivity
+mrp          / mrp.workorder            / workcenter_id (many2one)      : NEW relation: mrp.workcenter, required: required
+# NOTHING TO DO
+
+mrp          / procurement.order        / property_ids (many2many)      : DEL relation: mrp.property
+# NOTHING TO DO
+
+
+mrp          / stock.move               / consumed_for (many2one)       : DEL relation: stock.move
+# DONE: Pre-migration: Renamed to None and copied this column. That will be used in post-migration to known which moves are consumed for.
+
+mrp          / stock.move               / active_move_lot_ids (one2many): NEW relation: stock.move.lots
+mrp          / stock.move               / bom_line_id (many2one)        : NEW relation: mrp.bom.line
```

Comment:
This has to be filled as well, like also unit_factor https://github.com/OCA/OpenUpgrade/pull/1076#pullrequestreview-122561036

If this (or unit_factor) is not set, mrp.workorder record_production will NOT consume the move and it will cancel the move afterwards in button_mark_done, resulting in completely wrong quantities in stock for all the raw materials used in manufactoring orders that were created in the older version than v10..

Reply:
fill <code>bom_line_id</code> with what?

Reply:
Pretty much anything, otherwise as with unit_factor, the stock will go completely wrong when MO is marked as done. The (questionable) check only checks if it's set or not; if not set, it'll not increase quantity_done and end up cancelled in button_mark_done.

In my case I post-processed by checking through all BoM lines of the MO to find the same product - if only one is found (there could be multiple lines for same product), I assigned that to bom_line_id. But that will not solve all of them, as BoM might have changed in time, multiple same products found, etc. To finish it up, I just assigned what is left to basically any bom line of the MO's BoM - in practice it only matters if MO produced quantity is updated via wizard, and my customer doesn't use that. Multiple raw material stock moves of the same MO-s pointing at the same bom line is fine as well, with the same caveat of update wizard not working right. So it's under the understanding that MO Produced Qty Update wizard is not going to be used for old MOs, and then all is good.
That said, v10 code is still rather questionable - nothing stops you from removing a BoM line after it's already assigned to a stock.move, thus breaking the on hand quantities later on as well; for that an issue should be filed against Odoo and at the very least stock.move:bom_line_id made ondelete='restrict' instead of the current default SET NULL. I made that change in my own module for now with issue filing as TODO :(